csrc/host_ir/executor.h

```diff
@@ -47,20 +48,25 @@ class HostIrExecutor final : public OptInDispatch {
  public:
   HostIrExecutor(
       std::unique_ptr<HostIrContainer> container,
+      Communicator* communicator = nullptr,
       HostIrExecutorParams = HostIrExecutorParams());
-  std::vector<at::Tensor> runWithInput(const std::vector<c10::IValue>& inputs);
+  std::vector<at::Tensor> runWithInput(
+      std::unordered_map<Val*, c10::IValue> val_to_IValue);
 
  private:
   using OptInDispatch::handle;
   void handle(PostOnStream* post) override;
+  void postCompute(PostOnStream* post);
+  void postCommunication(PostOnStream* post);
```

Comment:
nit: I find this naming a little unintuitive. "post" in "postCompute" and "postCommunication" are a verb, but then it also appears as the parameter.  Maybe renaming the parameter to something "host_op"?

Reply:
right, I see what you mean. Wdyt about naming it `post_host_op`, or `post_ir` ?